lib/process/model/process/process_type.ex

```diff
@@ -7,7 +7,7 @@ defprotocol Helix.Process.Model.Process.ProcessType do
   @spec dynamic_resources(t) :: [resource]
   def dynamic_resources(data)
 
-  @spec event(t, Process.t, :created | :completed) :: [Helix.Event.t]
+  @spec event(t, Process.t, :created | :completed) :: [struct]
```

Comment:
About this part, it's funny:

- Runtime dependencies don't cause your code to need to be recompiled if the dependency is modified
- Compiletime dependencies do
- If you have a compiletime dependency that has a runtime-dependency, you indirectly has that dependency but since it could affect the code generated on compiletime, it works just like a compiletime-dependency and thus you have to recompile your module whenever that dependency (that was not compiletime-dep of anything) changes
- Protocols are based around macros, so they are compiletime dependencies of their implementations
- Since adding a type implies into runtime dependency and protocols are compiletime dependencies, this implies that the external referenced type now works like a compiletime dependency, thus, any time that any event handler was modified, so were all ProcessType implementations and, by consequence, any module that was depending on them on certain conditions.

tl;dr, before this change, any event handler change would cause:
```
_build/dev/lib/helix/ebin/ MODIFY Elixir.Helix.Process.Controller.Process.beam
_build/dev/lib/helix/ebin/ MODIFY Elixir.Helix.Process.Controller.TableOfProcesses.Allocator.Plan.beam
_build/dev/lib/helix/ebin/ MODIFY Elixir.Helix.Process.Model.Process.ProcessType.beam
_build/dev/lib/helix/ebin/ MODIFY Elixir.Helix.Event.Dispatcher.beam
_build/dev/lib/helix/ebin/ MODIFY Elixir.Helix.Process.Model.Process.ProcessType.Software.Encryptor.ProcessType.beam
_build/dev/lib/helix/ebin/ MODIFY Elixir.Helix.Process.Model.Process.ProcessType.Atom.beam
_build/dev/lib/helix/ebin/ MODIFY Elixir.Helix.Process.Model.Process.ProcessType.Software.Decryptor.ProcessType.beam
_build/dev/lib/helix/ebin/ MODIFY Elixir.Software.Encryptor.ProcessType.beam
_build/dev/lib/helix/ebin/ MODIFY Elixir.Software.Decryptor.ProcessType.beam
_build/dev/lib/helix/ebin/ MODIFY Elixir.Helix.Process.Model.Process.ProcessType.BitString.beam
_build/dev/lib/helix/ebin/ MODIFY Elixir.Helix.Process.Model.Process.ProcessType.Float.beam
_build/dev/lib/helix/ebin/ MODIFY Elixir.Helix.Process.Model.Process.ProcessType.Function.beam
_build/dev/lib/helix/ebin/ MODIFY Elixir.Helix.Process.Model.Process.ProcessType.Integer.beam
_build/dev/lib/helix/ebin/ MODIFY Elixir.Helix.Process.Model.Process.ProcessType.List.beam
_build/dev/lib/helix/ebin/ MODIFY Elixir.Helix.Process.Model.Process.ProcessType.Map.beam
_build/dev/lib/helix/ebin/ MODIFY Elixir.Helix.Process.Model.Process.ProcessType.PID.beam
_build/dev/lib/helix/ebin/ MODIFY Elixir.Helix.Process.Model.Process.ProcessType.Port.beam
_build/dev/lib/helix/ebin/ MODIFY Elixir.Helix.Process.Model.Process.ProcessType.Reference.beam
_build/dev/lib/helix/ebin/ MODIFY Elixir.Helix.Process.Model.Process.ProcessType.Tuple.beam
```